GVFS/GVFS.UnitTests/Mock/Common/MockLocalGVFSConfigBuilder.cs

```diff
@@ -0,0 +1,78 @@
+using GVFS.Common;
+using System.Collections.Generic;
+
+namespace GVFS.UnitTests.Mock.Common
+{
+    public class MockLocalGVFSConfigBuilder
+    {
+        private string defaultRing;
+        private string defaultUpgradeFeedUrl;
+        private string defaultUpgradeFeedPackageName;
+
+        private Dictionary<string, string> entries;
+
+        public MockLocalGVFSConfigBuilder(
+            string defaultRing,
+            string defaultUpgradeFeedUrl,
+            string defaultUpgradeFeedPackageName)
+        {
+            this.defaultRing = defaultRing;
+            this.defaultUpgradeFeedUrl = defaultUpgradeFeedUrl;
+            this.defaultUpgradeFeedPackageName = defaultUpgradeFeedPackageName;
+            this.entries = new Dictionary<string, string>();
+        }
+
+        public MockLocalGVFSConfigBuilder WithUpgradeRing(string value = null)
+        {
+            return this.With(GVFSConstants.LocalGVFSConfig.UpgradeRing, value ?? this.defaultRing);
+        }
+
+        public MockLocalGVFSConfigBuilder WithNoUpgradeRing()
+        {
+            return this.WithNo(GVFSConstants.LocalGVFSConfig.UpgradeRing);
+        }
+
+        public MockLocalGVFSConfigBuilder WithUpgradeFeedPackageName(string value = null)
+        {
+            return this.With(GVFSConstants.LocalGVFSConfig.UpgradeFeedPackageName, value ?? this.defaultUpgradeFeedPackageName);
+        }
+
+        public MockLocalGVFSConfigBuilder WithNoUpgradeFeedPackageName()
+        {
+            return this.WithNo(GVFSConstants.LocalGVFSConfig.UpgradeFeedPackageName);
+        }
+
+        public MockLocalGVFSConfigBuilder WithUpgradeFeedUrl(string value = null)
+        {
+            return this.With(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl, value ?? this.defaultUpgradeFeedUrl);
+        }
+
+        public MockLocalGVFSConfigBuilder WithNoUpgradeFeedUrl()
+        {
+            return this.WithNo(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl);
+        }
+
+        public MockLocalGVFSConfigBuilder With(string key, string value)
+        {
+            this.entries.Add(key, value);
+            return this;
+        }
+
+        public MockLocalGVFSConfigBuilder WithNo(string key)
```

Comment:
`With` and `WithNo` can be `private` since they are never used outside this class.